src/Servers/Kestrel/Core/src/Internal/Http3/Http3Connection.cs

```diff
@@ -305,93 +331,72 @@ private void UpdateStreamTimeouts(DateTimeOffset now)
                     Debug.Assert(streamDirectionFeature != null);
                     Debug.Assert(streamIdFeature != null);
 
-                    if (!streamDirectionFeature.CanWrite)
-                    {
-                        var context = CreateHttpStreamContext(streamContext);
-                        var streamType = await ReadNextStreamHeader(context);
+                    var context = CreateHttpStreamContext(streamContext);
+                    var pendingStream = new Http3PendingStream(context, streamDirectionFeature.CanWrite);
```

Comment:
You don't know it's a request stream until after you read the stream type. CanWrite could be true for any bidirectional stream.

Reply:
But they're not just control streams or request streams anymore, they could be webtransport uni/bidirectional streams as well. The Abort errors will need to be changed, I think you just need one general error about not being able to determine the stream type.